### PR TITLE
Make sure FSO still compiles on Linux

### PR DESCRIPTION
**Pull request by ngld**
This fixes a few issues clang complained about and adds the necessary code lib/ffmpeg/CMakeLists.txt.

I had to change the lib/lua/CMakeLists.txt file because CMake didn't find the correct include directory on my system. This change fixes it by using pkg-config instead of CMake's FindPackage function.

### CODE REVIEW

File: code/cutscene/ogg/OggDecoder.cpp [hunk 1]
```diff
@@ -233,7 +234,7 @@ namespace cutscene
                 vorbis_block_init(&movie.vstate, &movie.vblock);
             }
 
-            auto fps = static_cast<float>(movie.tinfo.fps_numerator) / movie.tinfo.fps_denominator;
+            fps = static_cast<float>(movie.tinfo.fps_numerator) / movie.tinfo.fps_denominator;
```
Comment (asarium) [inline code/cutscene/ogg/OggDecoder.cpp]: Why did you move the declaration to the beginning of the function? In-place declarations are a lot clearer IMO.

**Comment by asarium** [comment]
Thank you for these changes, after that one thing is handled I'll merge this.

File: code/cutscene/ogg/OggDecoder.cpp (hunk 1 above)
Comment (ngld) [inline code/cutscene/ogg/OggDecoder.cpp]: Your "goto Error;" skips this declaration. Clang (and gcc) abort with an error because of this.

There's no particular reason why I picked the beginning of the function, so feel free to move the declaration anywhere before the first use of goto.

Comment (asarium) [inline code/cutscene/ogg/OggDecoder.cpp]: I adapted the old Ogg loader and didn't bother cleaning it up. I'll remove the goto stuff which should fix the declaration issues.

**Comment by ngld** [comment]
Thanks for merging this quickly!
